chore(CHANGELOG): update to 5.1.0 (#3024)
camunda/camunda-modeler: CHANGELOG.md

___Note:__ Yet to be released changes appear here._

## 5.1.0

### Main changes

> * Multiselect context pad added to our tools for quick deletion, alignement and spacing

Reviewer: ```suggestion
* Multi-element context pad added to our tools for quick deletion, alignment, and spacing
```

> * Telemetry migrated to Mixpanel

Reviewer: Not sure if this should be mentioned here. Our users don't benefit from the change.

Reviewer 2: What we sometimes do is to add a one line summary in the beginning of the release (cf. https://github.com/bpmn-io/diagram-js/blob/develop/CHANGELOG.md#850). I'd encourage you to drop the "Main changes" section in favor of the existing pattern.

> * `bpmn-js` and `dmn-js` have received UX improvements

Reviewer: ```suggestion
* BPMN and DMN editors have received UX improvements:
```

* Conditional element template properties available in C8 
* Stability and QOL improvements across the application

### General

* `FEAT`: mask non-boolean flag values for ping event ([#2963](https://github.com/camunda/camunda-modeler/pull/2963))
* `FEAT`: track set flags via ping event ([#2963](https://github.com/camunda/camunda-modeler/pull/2963))
* `FEAT`: use @camunda/linting ([#2976](https://github.com/camunda/camunda-modeler/pull/2976))
* `FEAT`: implement mixpanel telemetry ([#2934](https://github.com/camunda/camunda-modeler/issues/2934), [fec6ee4](https://github.com/camunda/camunda-modeler/commit/fec6ee4))
* `FEAT`: track diagram deployments in mixpanel ([#2934](https://github.com/camunda/camunda-modeler/issues/2934), [a3e48b0](https://github.com/camunda/camunda-modeler/commit/a3e48b0))
* `FEAT`: track link and button clicks in mixpanel ([#2934](https://github.com/camunda/camunda-modeler/issues/2934), [bbf0914](https://github.com/camunda/camunda-modeler/commit/bbf0914))
* `FEAT`: track overlay usage in mixpanel ([#2934](https://github.com/camunda/camunda-modeler/issues/2934), [4e09d2e](https://github.com/camunda/camunda-modeler/commit/4e09d2e))

> * `FIX`: Debounced input fields no longer lose their values ([#2990](https://github.com/camunda/camunda-modeler/issues/2990))

Reviewer: Mixing lowercase and sentence-cased entries.

* `FIX`: Complete direct editing on selection changed ([#2961](https://github.com/camunda/camunda-modeler/issues/2961))
* `FIX`: Lint subscriptions only if start event is child to a subprocess ([#2983](https://github.com/camunda/camunda-modeler/issues/2983))
* `FIX`: cancel direct editing before shape deletion ([bpmn-js#1677](https://github.com/bpmn-io/bpmn-js/issues/1677))
* `DEPS`: bump camunda-bpmn-js to 0.15.2 ([bc8b74870](https://github.com/camunda/camunda-modeler/commit/bc8b74870), [dbe78ef45](https://github.com/camunda/camunda-modeler/commit/dbe78ef45))
* `DEPS`: bump bpmn-js-properties-panel to 1.2.0 ([bc8b74870](https://github.com/camunda/camunda-modeler/commit/bc8b74870))
* `DEPS`: bump bpmn-js to 9.3.1 ([bc8b74870](https://github.com/camunda/camunda-modeler/commit/bc8b74870))
* `DEPS`: bump bpmnlint-plugin-camunda-compat to 0.9.1 ([#2927](https://github.com/camunda/camunda-modeler/issues/2927), [d6cc308](https://github.com/camunda/camunda-modeler/commit/d6cc308))

### DMN

* `FEAT`: rework diagram interaction handles ([diagram-js#640](https://github.com/bpmn-io/diagram-js/pull/640))

> * `FEAT`: rework select and hover interaction on the diagram ([diagram-js#640](https://github.com/bpmn-io/diagram-js/pull/640), [diagram-js#643](https://github.com/bpmn-io/diagram-js/pull/643))

Reviewer: ```suggestion
* `FEAT`: rework select and hover interaction on the diagram ([diagram-js#640](https://github.com/bpmn-io/diagram-js/pull/640), [diagram-js#643](https://github.com/bpmn-io/diagram-js/pull/643))
* `FIX`: allow to scroll variable type in literal expression ([#2908](https://github.com/camunda/camunda-modeler/issues/2908))
```

___Note:__ Yet to be released changes appear here._

## 5.1.0

### Main changes

* Multi-element context pad added to our tools for quick deletion, alignment, and spacing
* Telemetry migrated to Mixpanel
* BPMN and DMN editors have received UX improvements:
	* clearer selection and multiselection
	* input fields in the properties panel no longer lose changes
* Conditional element template properties available in C8 

> * Stability and QOL improvements across the application

Reviewer: Not sure what QOL is. Also, is this about any specific changes? Let's make it explicit or remove it.
